models/login_source.go

```diff
@@ -403,6 +404,21 @@ func LoginViaLDAP(user *User, login, password string, source *LoginSource, autoR
 
 	var isAttributeSSHPublicKeySet = len(strings.TrimSpace(source.LDAP().AttributeSSHPublicKey)) > 0
 
+	// Update User admin flag if exist
+	isExist, err := IsUserExist(0, sr.Username)
+	if err != nil {
+		return nil, err
+	} else if isExist {
+		if !user.ProhibitLogin && source.IsSyncEnabled && len(source.LDAP().AdminFilter) > 0 && user.IsAdmin != sr.IsAdmin {
```

Comment:
```suggestion
		if !user.ProhibitLogin && len(source.LDAP().AdminFilter) > 0 && user.IsAdmin != sr.IsAdmin {
```

Reply:
from @guillep2k "...I still think this feature should be optional..." thats why @lafriks

Reply:
Not really as if you have set admin filter in ldap settings it should be followed not depending of syncing or not